File: python/242_Valid_Anagram.py
class Valid_Anagram:
    def isAnagram(self, s: str, t: str) -> bool:
        # Time complexity is O(n)
        if len(s) != len(t):
            return False
        countS, countT = {}, {}
        for char in s:
            countS[char] = countS.get(char, 0) + 1
        for char in t:
            countT[char] = countT.get(char, 0) + 1
        for x in countS:
            if countS.get(x) != countT.get(x):
                return False
        return True

File: c++/242_Valid_Anagram.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

class ValidAnagram {
public:
  bool isAnagram(std::string s, std::string t) {
    if (s.size() != t.size()){
      return false;
    }
    std::unordered_map <char,int> hashmap;
    for (int i = 0; i < s.size(); i++){
      hashmap[s[i]]++;
      hashmap[t[i]]--;
    }

    for (auto &count:hashmap){
      if (count.second != 0){
        return false;
      }
    }
    return true;
  }
};